Content.Shared/Damage/Components/DamageableComponent.cs

```diff
@@ -45,46 +46,56 @@ public class DamageableComponent : Component, IDamageableComponent, IRadiationAc
         public string DamageContainerId { get; set; } = DefaultDamageContainer;
 
         // TODO DAMAGE Cache this
-        [ViewVariables] public int TotalDamage => _damageList.Values.Sum();
-        [ViewVariables] public IReadOnlyDictionary<DamageGroupPrototype, int> DamageClasses => damageListToDamageGroup(_damageList);
-        [ViewVariables] public IReadOnlyDictionary<DamageTypePrototype, int> DamageTypes => _damageList;
+        [ViewVariables] public int TotalDamage => _damageDict.Values.Sum();
+        [ViewVariables] public IReadOnlyDictionary<DamageTypePrototype, int> DamagePerType => _damageDict;
+        [ViewVariables] public IReadOnlyDictionary<DamageGroupPrototype, int> DamagePerGroup => DamageGroupPrototype.DamageTypeDictToDamageGroupDict(_damageDict, ApplicableDamageGroups);
+        [ViewVariables] public IReadOnlyDictionary<DamageGroupPrototype, int> DamagePerSupportedGroup => DamageGroupPrototype.DamageTypeDictToDamageGroupDict(_damageDict, SupportedDamageGroups);
+
+        // TODO DAMAGE Cache this
+        // Whenever sending over network, need a <string, int> dictionary
+        public IReadOnlyDictionary<string, int> DamagePerGroupIDs => ConvertDictKeysToIDs(DamagePerGroup);
+        public IReadOnlyDictionary<string, int> DamagePerSupportedGroupIDs => ConvertDictKeysToIDs(DamagePerSupportedGroup);
+        public IReadOnlyDictionary<string, int> DamagePerTypeIDs => ConvertDictKeysToIDs(DamagePerType);
 
-        public HashSet<DamageGroupPrototype> SupportedGroups { get; } = new();
+        // TODO PROTOTYPE Replace these datafield variables with prototype referencess, once they are supported.
+        // Also requires appropriate changes in OnExplosion() and RadiationAct()
+        [ViewVariables]
+        [DataField("radiationDamageTypes")]
+        public List<string> RadiationDamageTypeIDs { get; set; } = new() {"Radiation"};
+        [ViewVariables]
+        [DataField("explosionDamageTypes")]
+        public List<string> ExplosionDamageTypeIDs { get; set; } = new() { "Piercing", "Heat" };
```

Comment:
why are these here in damageable? shouldn't these just be wherever explosion/radiation is dealing damage

Reply:
I have no idea. I think the logic was that different entities would be damaged by radiation or explosions in different ways maybe? but then the damage types were hardcoded?

I know some people were already talking about reworking radiation. For now, Im gonna leave them here, changing explosions & radiation can be done seperately